### PR TITLE
chore(clerk-js,nextjs,shared): Move devBrowser to shared

### PR DESCRIPTION
**Pull request by dimkl**
## Description

Move devBrowser utils to `@clerk/shared`. Ideally we should only expose some getters/setters to handle the devBrowser in cookie/query-parameter/localStorage, but currently since there is multiple ways to access the cookie and the localStorage is browser related i kept exposing some constants from shared and left some code handling the specific cases.

## Checklist

- [x] `npm test` runs as expected.
- [x] `npm run build` runs as expected.
- [ ] (If applicable) [JSDoc comments](https://jsdoc.app/about-getting-started.html) have been added or updated for any package exports
- [ ] (If applicable) [Documentation](https://github.com/clerk/clerk-docs) has been updated

## Type of change

- [ ] 🐛 Bug fix
- [ ] 🌟 New feature
- [ ] 🔨 Breaking change
- [x] 📖 Refactoring / dependency upgrade / documentation
- [ ] other:

## Packages affected

- [ ] `@clerk/backend`
- [ ] `@clerk/chrome-extension`
- [x] `@clerk/clerk-js`
- [ ] `@clerk/clerk-expo`
- [ ] `@clerk/fastify`
- [ ] `gatsby-plugin-clerk`
- [ ] `@clerk/localizations`
- [x] `@clerk/nextjs`
- [ ] `@clerk/clerk-react`
- [ ] `@clerk/remix`
- [ ] `@clerk/clerk-sdk-node`
- [x] `@clerk/shared`
- [ ] `@clerk/themes`
- [ ] `@clerk/types`
- [ ] `build/tooling/chore`

### CODE REVIEW

**Review by nikosdouvlis — APPROVED** [review]

File: packages/clerk-js/src/core/devBrowserHandler.ts [hunk 1]
```diff
@@ -157,13 +163,13 @@ export default function createDevBrowserHandler({
     if (devOrStgApi) {
       fapiClient.onBeforeRequest(request => {
         const devBrowserJWT = getDevBrowserJWT();
-        if (devBrowserJWT) {
-          request.url?.searchParams.set(DEV_BROWSER_SSO_JWT_PARAMETER, devBrowserJWT);
+        if (devBrowserJWT && request?.url) {
+          setDevBrowserJWTInURL(request.url, devBrowserJWT, true);
```
Comment (anagstef) [inline packages/clerk-js/src/core/devBrowserHandler.ts:167]: This function returns a value!!

File: packages/clerk-js/src/core/devBrowserHandler.ts [hunk 2]
```diff
@@ -157,13 +163,13 @@ export default function createDevBrowserHandler({
...
         }
       });
 
       fapiClient.onAfterResponse((_, response) => {
-        const newDevBrowserJWT = response?.headers?.get(DEV_BROWSER_SSO_JWT_HTTP_HEADER);
+        const newDevBrowserJWT = getDevBrowserJWTFromResponse(response);
         if (newDevBrowserJWT) {
           setDevBrowserJWT(newDevBrowserJWT);
```
Comment (anagstef) [inline packages/clerk-js/src/core/devBrowserHandler.ts:174]: Same as above

**Review by anagstef — CHANGES_REQUESTED** [review]

File: packages/clerk-js/src/core/devBrowserHandler.ts (hunk 2 above)
Comment (dimkl) [reply packages/clerk-js/src/core/devBrowserHandler.ts:174]: This is not the same method (the previous method was `setDevBrowserJWTInURL`)

File: packages/clerk-js/src/core/devBrowserHandler.ts (hunk 1 above)
Comment (dimkl) [reply packages/clerk-js/src/core/devBrowserHandler.ts:167]: nice catch! Fixed!

**Review by anagstef — APPROVED** [review]
💯

**Pull request merged**